### DISCUSSION TITLE
Searching addresses / Старбеево / Kurkino District

### DISCUSSION THREAD

**Discussion by Grass-snake-89** [osm-search/Nominatim#2399] [Geocoding around the world]
Hello, I ran into difficulty finding the address in Nominatim:

1. The following address tags have been added to the OSM:

https://www.openstreetmap.org/way/963467794
addr:allotments=СНТ «Восход-7»
addr:housenumber=111
addr:postcode=141401
addr:suburb=Старбеево

So there is no street in address.
If I search "химки восход-7 111", Nominatim finds the correct address (2nd line), but adds the nearest street (highway:residential=улица Ленина) to the output-address result. Which tags are correct to indicate the needed address?
https://nominatim.openstreetmap.org/ui/search.html?q=%D1%85%D0%B8%D0%BC%D0%BA%D0%B8+%D0%B2%D0%BE%D1%81%D1%85%D0%BE%D0%B4-7+111

2. I do a reverse search or select a line:

https://nominatim.openstreetmap.org/ui/details.html?osmtype=W&osmid=48869971
This building is inside the boundary "Kurkino District"
https://www.openstreetmap.org/relation/442741

But Nominatim finds the wrong boundary "North Tushino District":
https://www.openstreetmap.org/relation/445279
boundary:administrative (suburb)=район Северное Тушино

Tags of these two areas are the same and I don't understand the algorithm by which Nominatim correlates boundaries.
Please help me to fix this problems. Probaably I should change some tags in OSM.

**Comment by mtmail** [comment]
On (1) I think `addr:place=Старбеево` should work

"The addr:place tag is used for addresses where the house number does not refer to a street. A classic example are small villages that do not have street names but instead just sequentially number their houses." https://www.nominatim.org/2020/10/07/about-addr-tags.html

On (2) Nominatim looks at the street to find the administrative boundary. The OSM way https://nominatim.openstreetmap.org/ui/details.html?osmtype=W&osmid=532579227&class=highway is in North Tushino District. I'm not saying that's correct or that Nominatim does the right thing here, I'm just explaining the logic/algorihmn.

**Comment by Grass-snake-89** [comment]
Thanks for the answer. You've made a good engine for finding an address, but (2) sometimes it doesn't work as expected. I really hope that the situation will change soon.
